domain/application/state/application.go

```diff
@@ -638,9 +647,48 @@ WHERE name = $unitDetails.name
 	return unit.LifeID, errors.Annotatef(err, "querying unit %q life", unitName)
 }
 
-// ApplicationScaleState looks up the scale state of the specified application, returning an error
+// SetUnitLife sets the life of the specified unit, returning an error
+// satisfying [applicationerrors.UnitNotFound] if the unit is not found.
+func (st *ApplicationState) SetUnitLife(ctx domain.AtomicContext, unitName string, l life.Life) error {
+	unit := unitID{Name: unitName, LifeID: l}
+	query := `
+SELECT uuid AS &unitID.uuid
```

Comment:
We can wrap this into a helper function and reuse it.

Reply:
we only use this in this place right now, can do later if needed